pkg/minibroker/postgres.go

```diff
@@ -34,72 +45,70 @@ func (p PostgresProvider) Bind(services []corev1.Service, params map[string]inte
 
 	host := buildHostFromService(service)
 
-	dbVal, ok := params["postgresqlDatabase"]
-	if !ok {
+	database, err := provisionParams.DigStringAltOr(
 		// Some older chart versions use postgresDatabase instead of postgresqlDatabase.
-		dbVal, ok = params["postgresDatabase"]
-		if !ok {
-			dbVal = ""
-		}
+		[]string{"postgresqlDatabase", "postgresDatabase"},
+		"",
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get database name: %w", err)
 	}
-	database, ok := dbVal.(string)
-	if !ok {
-		return nil, errors.Errorf("database not a string")
-	}
-
-	userVal, ok := params["postgresqlUsername"]
-	if !ok {
+	user, err := provisionParams.DigStringAltOr(
 		// Some older chart versions use postgresUsername instead of postgresqlUsername.
-		userVal, ok = params["postgresUsername"]
-		if !ok {
-			userVal = "postgres"
-		}
-	}
-	user, ok := userVal.(string)
-	if !ok {
-		return nil, errors.Errorf("username not a string")
+		[]string{"postgresqlUsername", "postgresUsername"},
+		defaultPostgresqlUsername,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get username: %w", err)
 	}
 
-	var password string
-	if user != "postgres" {
-		// postgresql-postgres-password is used when postgresqlPostgresPassword is set and
-		// postgresqlUsername is not 'postgres'.
-		passwordVal, ok := chartSecrets["postgresql-postgres-password"]
-		if !ok {
-			passwordVal, ok = chartSecrets["postgresql-password"]
-			if !ok {
-				return nil, errors.Errorf("password not found in secret keys")
-			}
-		}
-		password, ok = passwordVal.(string)
-		if !ok {
-			return nil, errors.Errorf("password not a string")
-		}
+	var passwordKey, altPasswordKey string
+	// postgresql-postgres-password is used when postgresqlPostgresPassword is set and
+	// postgresqlUsername is not 'postgres'.
+	if _, ok := provisionParams.Dig("postgresqlPostgresPassword"); ok && user != defaultPostgresqlUsername {
+		passwordKey = "postgresql-postgres-password"
 	} else {
-		passwordVal, ok := chartSecrets["postgresql-password"]
-		if !ok {
-			// Chart versions <2.0 use postgres-password instead of postgresql-password.
-			// See https://github.com/kubernetes-sigs/minibroker/issues/17
-			passwordVal, ok = chartSecrets["postgres-password"]
-			if !ok {
-				return nil, errors.Errorf("password not found in secret keys")
+		passwordKey = "postgresql-password"
+		// Chart versions <2.0 use postgres-password instead of postgresql-password.
+		// See https://github.com/kubernetes-sigs/minibroker/issues/17
+		altPasswordKey = "postgres-password"
+	}
+	password, err := chartSecrets.DigString(passwordKey)
+	if err != nil {
+		switch err {
+		case ErrDigNotFound:
+			password, err = chartSecrets.DigString(altPasswordKey)
```

Comment:
Shouldn't this be using `DigOr`?

Reply:
Yes! Actually, `DigStringAlt`.

Reply:
Fixed with https://github.com/kubernetes-sigs/minibroker/pull/99/commits/b034596285a5e7b3d2305ff145012f956889d95a.